CMakeLists.txt

```diff
@@ -3,6 +3,17 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+# List of all partitions supported by TF-M
+# Name must match name in 'trusted-firmware-m/config/config_default.cmake'
+set(tfm_partitions
+  TFM_PARTITION_PROTECTED_STORAGE
+  TFM_PARTITION_INTERNAL_TRUSTED_STORAGE
+  TFM_PARTITION_CRYPTO
+  TFM_PARTITION_INITIAL_ATTESTATION
+  TFM_PARTITION_PLATFORM
+  TFM_PARTITION_AUDIT_LOG
+  )
```

Comment:
Isn't that a bit overkill? If we do as you say elsewhere, and we set the default to OFF, then the only thing we have to do if a new partition is added is to add the entry in the list during the upmerge. I thought that we wanted to avoid unnecessary `execute_process` to make the execution time of the configure stage faster.

Some synchronization would still have to be done, since the name of the existing partitions might change.

Reply:
> Some synchronization would still have to be done, since the name of the existing partitions might change.

Right, especially on the Kconfig side, as you are using the same symbols there.
I'm fine with keeping a list, only proposed automation, to minimize the burden on keeping the names in sync for the module maintainers.

Reply:
No need to change.